### ISSUE TITLE
Removing a 'complete' event listener in a handler can cause later transactions not to complete

### ISSUE THREAD

**Issue by quasicomputational** [dumbmatter/fakeIndexedDB#26]
Reproduction:

```js
import indexedDB from "fake-indexeddb";

export const STORE = "store";

export const checkForGarbage = () => {
    const dbReq = indexedDB.open("db", 1);
    dbReq.addEventListener("upgradeneeded", (ev) => {
        ev.target.result.createObjectStore(STORE, { autoIncrement: true });
    });
    dbReq.addEventListener("success", (ev) => {
        const db = ev.target.result
        const tx = db.transaction([STORE], "readwrite");
        const complete = () => {
            tx.removeEventListener('complete', complete);
        };
        tx.addEventListener('complete', complete);
        const store = tx.objectStore(STORE);
        store.get(0).addEventListener("success", () => console.log("Done!"));
    });
};

for (let i = 0; i < 5; i++) {
    checkForGarbage();
}
```

The expected result is to print "Done!" five times. However, only two are produced. Commenting out the body of `complete` (i.e., removing the `removeEventListener` call) produces the desired result.

**Comment by dumbmatter** [comment]
Want me to make a new release, to get this bug fix out? I figure I might as well do that, unless you were planning some other fixes in the very near future.

**Comment by quasicomputational** [comment]
Any time would be convenient, thanks. This bug isn't even biting me at the moment, as far as I know, but it will be once `idb` 4.0.0 comes out and I want to upgrade. I don't have anything more queued up imminently - maybe a look at the new WPT failures at some point and seeing if there are easy wins.

**Comment by dumbmatter** [comment]
Okay, I just published v2.0.6 :)

Out of curiosity, what about idb 4 would trigger this, something internal to that library?

**Comment by quasicomputational** [comment]
Yeah, basically doing things different internally - it's being more systematic about how it attaches listeners and cleans them up afterwards now. I looked closely at its code because it was the thing that'd changed to make my tests red, but it's not doing anything weird.
